teps/0089-spire-support.md

```diff
@@ -0,0 +1,154 @@
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A zero-trust supply chain for Tekton requires collaboration between Tekton Pipelines, Tekton Chains, and [SPIRE](https://spiffe.io/).
+A detailed explanation of why this is necessary can be found in [Zero Trust Supply Chains](https://docs.google.com/document/d/1CRvANkYu0fxJjEZO4KTyyk_1uZm2Q9Nr0ibxplakODg/edit?resourcekey=0-nGnWnCni8IpiXim-WreYMg#heading=h.fyy27kd27z1r) by dlorenc@.
+
+To summarize, we can't rely on only Tekton Pipelines and Tekton Chains to secure our pipeline because Chains relies on Pipelines to tell it what the given state of a TaskRun is.
+For example, Chains expects Pipelines to tell it when the TaskRun is complete, what happened in the TaskRun, and what artifacts need to be signed.
+This is insecure because any cluster administrator can edit a TaskRun to say anything they want, and Chains will just believe it.
+Chains has no way to verify that the TaskRun it gets was actually run as it says it was.
+
+This is where SPIRE comes in! 
+SPIRE can be used to attest workloads on a cluster, and in this case it will be used to attest TaskRuns.
+Together, Tekton Pipelines, Tekton Chains, and SPIRE can provide a secure zero-trust supply chain on Kubernetes.
+
+
+## Motivation
+
+Security! =)
+
+We'll also need this feature for Tekton to achieve [SLSA 3](https://slsa.dev/levels), which requires Non-falsifiable Provenance.
+
+### Goals
+
+* Pipelines should attest a TaskRun against SPIRE before it completes
+* Pipelines should store info necessary for verification somewhere
+* Chains should be able to verify a TaskRun wasn't modified before signing
+
+
+## Proposal
+The basic design looks like this:
+
+[TODO: insert diagram here]
+
+1. Entrypoint image creates a Pod for a new TaskRun
+1. Entrypoint image is modified to request a signature from SPIRE over the TaskRun yaml
```

Comment:
does the entrypoint do this, or is this maybe the Pipelines controller (image)?

wondering b/c the tekton pipelines controller is usually responsible for a) seeing a taskrun, b) creating a pod for that task run, and THEN c) the entrypoint image/entrypointer become involved once the steps start executing